### PR TITLE
feat: VRF support

### LINKED ISSUE #268: Add support for VRF features
### What problem are you facing?

VRF features are being added to the IP, Gateway, and Virtual Circuit resources in the Equinix Metal API.
https://deploy.equinix.com/developers/docs/metal/interconnections/vrf/

There is no way to access these features in the Metal CLI today.

### How could the Equinix Metal CLI help solve this problem?

The existing `metal gateway` and `metal ip` action groups should learn new actions and add relevant fields based on VRF capabilities.

Packngo support for VRF was initially added here: https://github.com/packethost/packngo/issues/325

Metal Go support is present (and more current than packngo's support): https://github.com/search?q=repo%3Aequinix-labs%2Fmetal-go%20vrf&type=code

Examples of how this could be implemented are in @goldenprifti's Equinix Metal VRF CLI gist here: https://gist.github.com/goldenprifti/850fa74cb08c23b4e3ebac2b01746bab

These features should also be added to a `metal interconnection` action group (which does not exist today #307).

### PR DESCRIPTION
**Pull request by irusoeqx**
Hi guys,

Apologies that this took so long, but I was doing a bunch other things, but here I am committing the code that Marques fixed up along with some other functions, delete and list, I hope this is an alright starting point for whoever works on this next.

### CODE REVIEW

**Review by cprivitere — DISMISSED** [review]

**Comment by displague** [comment]
Hi, @irusoeqx! Thanks for following up on #271.

It looks like a portion of the PR is missing compared to https://github.com/equinix/metal-cli/compare/main...displague:metal-cli:irusoeqx/main?diff=unified (specifically, cli.go, go.sum, and go.mod updates)

You can also see there are a number of linter and compile time errors highlighted in the file view:
https://github.com/equinix/metal-cli/pull/358/files

Try running the code and each action that has been added before pushing the next update, from the project root:

```
$ go run ./cmd/metal vrf help
$ go run ./cmd/metal vrf create --the-required-args
$ go run ./cmd/metal vrf list --the-required-args
$ go run ./cmd/metal vrf delete --the-required-args
```

**Comment by cprivitere** [comment]
Hey @irusoeqx I rebased and ran the e2e tests for the current version of your PR, but it looks like the tests are still failing with various issues.

File: internal/vrf/delete.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
+package main // or your appropriate package name
```
Comment (displague) [inline internal/vrf/delete.go]: this should match the other files in the directory
```suggestion
package vrf
```

**Comment by codinja1188** [comment]
TC: https://github.com/equinix/metal-cli/pull/377

**Comment by aayushrangwala** [comment]
@irusoeqx This PR https://github.com/equinix/metal-cli/pull/377 aims to add the same support. Please feel free to further contribute or close the PR in the favor of other

**Comment by displague** [comment]
This PR became the basis for #377. Closing.

**Pull request closed without merging**